### PR TITLE
chore(ts-migration): adds api extrator for public definitions files and strip Internals

### PR DESCRIPTION
**Pull request by nunomaduro**
This pull request **now** uses api extractor for validating the public files we are exporting instead of a single file.

Here is an example:

Screenshot 2020-04-08 at 14 47 41

In addition, it adds **stripInternals** option the tsconfig file, so we can hide some internals from the public API.

### CODE REVIEW

**Comment by Haroenv** [comment]
Did you think the warnings from api-extractor weren't useful after all?

**Comment by nunomaduro** [comment]
@Haroenv They are useful - but the goal of having API Extrator is detect missing exported types on a a **single file public api**. With the current approach, and to move faster, we are going to have around 120 files.

**Comment by nunomaduro** [comment]
@Haroenv Maybe in the future, and with more time, once we have root level exports and can have again api extrator 👍

**Comment by francoischalifour** [comment]
Will this configuration be enough to catch error types like when `InstantSearchOptions` is not exported? If not, how important do you think it is?

**Comment by nunomaduro** [comment]
> Will this configuration be enough to catch error types like when InstantSearchOptions is not exported? If not, how important do you think it is?

You just gave me a wonderful idea. Give me a second.

File: package.json [hunk 1]
```diff
@@ -23,9 +23,9 @@
     "start": "yarn run dev",
     "dev": "yarn run storybook",
     "build": "yarn run build:cjs && yarn run build:es && yarn run build:umd && yarn run build:types",
-    "build:umd": "BABEL_ENV=umd rollup --config scripts/rollup/config.js",
```
Comment (nunomaduro) [inline package.json:26]: Prevents builds of keep trash files from previous builds.

File: src/widgets/geo-search/geo-search.js [hunk 2]
```diff
@@ -93,7 +93,6 @@ const suit = component('GeoSearch');
  *
  * Don't forget to explicitly set the `height` of the map container (default class `.ais-geo-search--map`), otherwise it won't be shown (it's a requirement of Google Maps).
  *
- * @type {WidgetFactory}
```
Comment (nunomaduro) [inline src/widgets/geo-search/geo-search.js:96]: API Extrator can't find this type, really don't know why. This file will be migrated to typescript anyways.

File: tsconfig.declaration.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "./tsconfig",
   "compilerOptions": {
+    "stripInternal": true,
```
Comment (nunomaduro) [inline tsconfig.declaration.json:4]: Prevents internals of being exposed.

**Comment by nunomaduro** [comment]
cc @Haroenv @francoischalifour pull request updated.

**Pull request merged**